src/sections/dashboard/sidebar/VerticalSidebar.js

```diff
@@ -1,3 +1,312 @@
+import DashboardMode, {
+  DASHBOARD_MODE,
+} from '~/layout/dashboard/header/DashboardMode';
+import { useAppDispatch, useAppSelector } from '~/redux/hook';
+import { setDashboardType } from '~/redux/slices/appSlice';
+
+const DRAWER_WIDTH = 250;
+const DRAWER_WIDTH_COLLAPSED = 70;
+
 export default function VerticalSidebar() {
-  return <div>VerticalSidebar</div>;
+  const { dashboardType } = useAppSelector((store) => store.app);
+  const displayedType = DASHBOARD_TYPE.find(
+    (item) => item.id === dashboardType,
+  );
+  const router = useRouter();
+  const pathname = usePathname();
+  const t = useTranslations();
+  const dispatch = useAppDispatch();
+
+  const [isOpen, setIsOpen] = useState(true);
+
+  const otherItems = useMemo(
+    () => displayedType.children.filter((item) => item.label !== 'Setting'),
+    [displayedType],
+  );
+  const settingsItem = useMemo(
+    () => displayedType.children.find((item) => item.label === 'Setting'),
+    [displayedType],
+  );
+
+  useEffect(() => {
+    const foundType = DASHBOARD_TYPE.find((type) =>
+      pathname.includes(type.path),
+    );
+    if (foundType) {
+      dispatch(setDashboardType(foundType?.id));
+    }
+  }, [pathname]);
+
+  return (
+    <Box sx={{ position: 'relative' }}>
+      <DrawerStyled variant="permanent" anchor="left" open={isOpen}>
+        <Stack
+          sx={{
+            justifyContent: 'space-between',
+            flexGrow: 1,
+          }}
+        >
+          <Box
+            sx={{
+              height: '50px',
+              display: 'flex',
+              alignItems: 'center',
+              marginBottom: '40px',
+              paddingX: '16px',
+              paddingTop: '16px',
+              justifyContent: 'space-between',
+            }}
+          >
+            <Image
+              src={
+                isOpen
+                  ? '/images/Logo_min.png'
+                  : '/images/ic-viet-qr-small-trans.svg'
+              }
+              width={58}
+              height={24}
+              priority
+              quality={100}
+              alt="VietQR Logo"
+              onClick={() => router.push('/')}
+            />
+            {isOpen && <DashboardMode mode={DASHBOARD_MODE.VERTICAL} />}
+          </Box>
+          <Box sx={{ flexGrow: 1 }}>
+            <List dense disablePadding>
+              {otherItems.map((item) => (
+                <ListItem
+                  key={item.id}
+                  disablePadding
+                  sx={{ paddingBottom: '8px' }}
+                >
+                  {item.children.length > 0 ? (
+                    <Box>
+                      {/* Parent Item */}
+                      {isOpen && (
+                        <Typography
+                          sx={{
+                            fontSize: {
+                              xs: '10px',
+                              md: '12px',
+                            },
+                            color: '#666A72',
```

Comment:
same [this](https://github.com/duchieupham/vietqr_web_2.0/pull/70/files#r1781219305)